### ISSUE TITLE
Allow users to increase size of more UI elements

### ISSUE THREAD

**Issue by dmalan** [sqlitebrowser/sqlitebrowser#2306]
Describe the new feature
--------------------------
We often use DB Browser to introduce students to SQL in classes. When demoing DB Browser in person, it's quite easy to zoom in on the UI (as via macOS's built-in Accessibility controls) so that students can see aspects of the UI more easily on a projector screen.

But for online classes (both now and also before the pandemic), it's harder for students to see aspects of the UI, particularly, even when sharing just DB Browser's window via, e.g., Zoom. If an instructor zooms in on the UI on their machine (as via macOS's same built-in Accessibility controls), the zooming only happens on their side; what students see via the shared screen is unchanged. A workaround is for the instructor to change their resolution (e.g., to 720p or even lower), but that then sacrifices quite a bit of screen real estate on their end. 

We do mitigate this somewhat by increasing DB Browser's font size, but that only seems to affect the cells under **Browse Data**. The **Database Structure** tab (where we spend a good amount of time in class) remains unchanged. By contrast, browsers make it quite easy to grow all UI elements at once, as via command-minus or -plus.

I suspect this is easier said than done UI-wise, given DB Browser's cross-platform design, but if there were a mechanism to magnify the size of all UI elements, that would be terrific for class. Alternatively, even just allowing users to increase the font size of the text in the **Database Structure** tab would be a big help.

Thank you for a wonderful teaching tool!

Additional info
---------------
Please answer these questions before submitting your feature request.

#### Is your feature request related to an issue? Please include the issue number.

Similar in spirit to https://github.com/sqlitebrowser/sqlitebrowser/issues/2016 but a more generalized idea.

#### Does this feature exist in another product or project?

E.g., Chrome.

##### default font size

<img width="1081" alt="a copy" src="https://user-images.githubusercontent.com/788678/85881784-2fa7b780-b7ac-11ea-94b1-7cf28e9f5578.png">

<img width="1081" alt="b copy" src="https://user-images.githubusercontent.com/788678/85881793-32a2a800-b7ac-11ea-9782-c5889a17f562.png">

##### with font size increased

<img width="755" alt="Screen Shot 2020-06-25 at 9 22 47 AM" src="https://user-images.githubusercontent.com/788678/85881444-9c6e8200-b7ab-11ea-8eb2-f1380f232d6b.png">

<img width="1081" alt="d copy" src="https://user-images.githubusercontent.com/788678/85881810-3a624c80-b7ac-11ea-8818-efbca0b9b198.png">

**Comment by justinclift** [comment]
Yeah, we're probably overdue to go through the Preferences tab and make sure we have a setting there for every element in the UI.

It's probably good timing to raise this now, as @MKleusberg is thinking through how to redesign the UI anyway (eg allow multiple browsing panes, etc) in #2283. :smile:

**Comment by MKleusberg** [comment]
@dmalan Would a general option for increasing the size of all text do? That would be very easy to implement but obviously enlarge not only the more important text e.g. in the Database Structure view but also all the menu items.

**Comment by dmalan** [comment]
Hi @MKleusberg, @justinclift, thanks so much for considering. I imagine that would do the trick, @MKleusberg, so long as it wouldn't break any UI elements (e.g., buttons or column headings or such). For our own classroom use, we pretty much spend all of our time using:

* **Database Structure** (including **Create Table** and **Create Index**)
* **Browse Data** (including drawing students' attention to the row count along the bottom)
* **Execute SQL** (including **Open tab** and **Execute all/selected SQL**)

So anything that scales all text visible in those UI panels would likely prove a big help! Thanks so much.

**Comment by MKleusberg** [comment]
Hi @dmalan! Can you give the next nightly build a try? You can download it [here](https://nightlies.sqlitebrowser.org/latest/) tomorrow. It adds two more options to the Preferences dialog. The first is ```Database structure font size``` which you can find in the ```Database``` tab and which changes the font size of the tree view in the Database Structure tab. The second is the general ```Font size``` which you can find in the ```General``` tab which changes the font size of all elements in the UI for which there are no other font size options.

So in theory you should be able to the the structure, data, and SQL font sizes separately to some value which suits your needs. And if needed you can then increase the font size of the rest of the UI a bit too. Let me know if it works for you or if you want to configure anything else :smile:

**Comment by nadler** [comment]
I was also interested in increasing font size, in particular the font in the results log of Execute SQL as that remains small after increasing all the fonts I can.  I tried to run the Mac OS nightly build from Jul 24 23:18 but the OS said it was corrupt.

**Comment by justinclift** [comment]
@nadler Would you be ok to screenshot the error message about that corruption?

We should be able to fix it, but need to know the exact details. :smile:

**Comment by dmalan** [comment]
Hi @MKleusberg, so sorry for the belated reply. The new option is great. Thanks so much. Though did the default font sizes perhaps change since that option's introduction? As best I can tell, it looks like the default font size for **Database Structure** and **Browse Data** is now smaller (10pt)? Might it make sense for the default to be closer to, e.g., 14pt, to match the other UI elements?

10pt currently:

<img width="1356" alt="Screen Shot 2020-07-26 at 11 12 06 AM" src="https://user-images.githubusercontent.com/788678/88482771-88947800-cf31-11ea-9189-9c3483c88622.png">

14pt, for example:

<img width="1356" alt="Screen Shot 2020-07-26 at 11 14 58 AM" src="https://user-images.githubusercontent.com/788678/88482777-90541c80-cf31-11ea-8a71-0aedc6003593.png">

**Comment by nadler** [comment]
> @nadler Would you be ok to screenshot the error message about that corruption?
> 
> We should be able to fix it, but need to know the exact details. 😄

I tried again with the Jul 30 build and it ran.  And even better, the font changes are fantastic!  Thanks so much!

**Comment by justinclift** [comment]
Awesome. :smile:

**Comment by nadler** [comment]
Using the Jul 30 build some more, one or more of these font size settings don't seem to save.  After adjusting them I saw the font sizes change.  Then quitting and restarting the application still shows the values I changed, but I'm not seeing fonts in that size.  The first picture is the text after I've changed my preferences.  The second is after restarting the application.

<img width="438" alt="AfterSetting" src="https://user-images.githubusercontent.com/5439597/89335742-09ecb880-d65e-11ea-886c-8d9a9d8b1949.png">
<img width="268" alt="AfterRestart" src="https://user-images.githubusercontent.com/5439597/89335764-0fe29980-d65e-11ea-98dd-cabe9ca2e7ab.png">

**Comment by mgrojo** [comment]
@nadler Are you using macos? I think some people reported a problem in saving preferences. Could you check the value of the font sizes in the Preferences dialog?

**Comment by nadler** [comment]
> @nadler Are you using macos? I think some people reported a problem in saving preferences. Could you check the value of the font sizes in the Preferences dialog?

I am using Mac OS X (10.15.6).  The value was correct in the preferences dialog after the restart, but it wasn't using the value that was shown in the preferences.  Modifying the value again caused it to work.

**Comment by justinclift** [comment]
Ahhhh.  That sounds like there's a bug in our code where the saved preferences are first loaded from disk then applied.
